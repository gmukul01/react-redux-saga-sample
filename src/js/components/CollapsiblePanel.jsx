import React, { PureComponent } from "react";
import uniqid from "uniqid";
import PropTypes from "prop-types";

export default class CollapsiblePanel extends PureComponent {
  render() {
    const targetId = uniqid();
    return (
      <div className="panel panel-default">
        <div className="panel-heading" data-toggle="collapse" data-target={`#${targetId}`}>
          <h5 className="panel-title">{this.props.title}</h5>
        </div>
        <div id={targetId} className="panel-collapse collapse">
          <div className="panel-body">{this.props.children}</div>
        </div>
      </div>
    );
  }
}
CollapsiblePanel.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
};
