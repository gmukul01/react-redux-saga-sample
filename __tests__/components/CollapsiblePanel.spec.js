import React from "react";
import uniqid from "uniqid";
import ShallowRenderer from "react-test-renderer/shallow";

import CollapsiblePanel from "components/CollapsiblePanel";

jest.mock("uniqid");
uniqid.mockImplementation(() => 42);

describe("CollapsiblePanel", () => {
  it("should render correctly", () => {
    const props = {
        title: "Dummy",
        children: <p>Dummy Children</p>
      },
      renderer = new ShallowRenderer(),
      result = renderer.render(<CollapsiblePanel {...props} />);
    expect(result).toMatchSnapshot();
  });
});
