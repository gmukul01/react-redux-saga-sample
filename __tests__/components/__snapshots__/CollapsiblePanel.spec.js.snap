// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsiblePanel should render correctly 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
    data-target="#42"
    data-toggle="collapse"
  >
    <h5
      className="panel-title"
    >
      Dummy
    </h5>
  </div>
  <div
    className="panel-collapse collapse"
    id={42}
  >
    <div
      className="panel-body"
    >
      <p>
        Dummy Children
      </p>
    </div>
  </div>
</div>
`;
